securesystemslib/metadata.py

```diff
@@ -0,0 +1,218 @@
+"""Dead Simple Signing Envelope
+"""
+
+import logging
+from typing import Any, Dict, List
+
+from securesystemslib import exceptions, formats
+from securesystemslib.serialization import (
+    BaseDeserializer,
+    BaseSerializer,
+    JSONDeserializer,
+    JSONSerializable,
+    JSONSerializer,
+    SerializationMixin,
+)
+from securesystemslib.signer import Key, Signature, Signer
+from securesystemslib.util import b64dec, b64enc
+
+logger = logging.getLogger(__name__)
+
+
+class EnvelopeJSONDeserializer(JSONDeserializer):
+    """Deserializes raw bytes and creates an Envelope object using JSON
+    Deserialization."""
+
+    def deserialize(self, raw_data: bytes) -> "Envelope":
+        """Deserialize utf-8 encoded JSON bytes into an instance of Envelope.
+
+        Arguments:
+            raw_data: A utf-8 encoded bytes string.
+
+        Raises:
+            DeserializationError: If fails to deserialize raw_data.
+
+        Returns:
+            dict.
+        """
+        try:
+            return Envelope.from_dict(super().deserialize(raw_data))
+        except Exception as e:
+            raise exceptions.DeserializationError(
+                "Failed to create Envelope"
+            ) from e
+
+
+class Envelope(SerializationMixin, JSONSerializable):
+    """DSSE Envelope to provide interface for signing arbitrary data.
+
+    Attributes:
+        payload: Arbitrary byte sequence of serialized body.
+        payload_type: string that identifies how to interpret payload.
+        signatures: list of Signature.
+
+    """
+
+    def __init__(
+        self, payload: bytes, payload_type: str, signatures: List[Signature]
+    ):
+        self.payload = payload
+        self.payload_type = payload_type
+        self.signatures = signatures
```

Comment:
Yeah, I think we kept it a list for simplicity, and because that's what the DSSE spec describes. I'm fine with using a dict internally, if there's a clear usage advantage. Though I don't know if the same requirements apply as for TUF `Metadata`, since I expect `Envelope` objects to usually have a much shorter lifetime (also see https://github.com/theupdateframework/python-tuf/pull/2246#pullrequestreview-1253567186).

Reply:
Any opinions about this, @PradyumnaKrishna?